### PR TITLE
#10235. Retention time for history

### PR DESCRIPTION
**Pull request by jgandres**

### CODE REVIEW

File: include/mega/megaclient.h [hunk 1]
```diff
@@ -689,6 +689,9 @@ class MEGA_API MegaClient
     // get welcome pdf
     void getwelcomepdf();
 
+    // set retention time for a chatroom, after which all messages in the chat are automatically deleted
```
Comment (sergiohs84) [inline include/mega/megaclient.h:692]:
```suggestion
    // set retention time for a chatroom, after which older messages in the chat are automatically deleted
```

File: include/megaapi.h [hunk 2]
```diff
@@ -15633,6 +15633,30 @@ class MegaApi
          */
         void archiveChat(MegaHandle chatid, int archive, MegaRequestListener *listener = NULL);
 
+        /**
+         * @brief Allows a logged in operator/moderator to specify a message retention timeframe
+         * after which all messages in the chat are automatically deleted.
```
Comment (sergiohs84) [inline include/megaapi.h:15638]:
```suggestion
         * after which older messages in the chat are automatically deleted.
```

File: include/megaapi.h [hunk 3]
```diff
@@ -15633,6 +15633,30 @@ class MegaApi
...
+         * - MegaRequest::getNumdetails - Returns the retention timeframe
+         * - MegaRequest::getFlag - Returns true if period is specified in seconds, otherwise returns false.
+         *
+         * On the onRequestFinish error, the error code associated to the MegaError can be:
+         * - MegaError::API_EARGS - If the chatid is invalid
+         * - MegaError::API_ENOENT - If there isn't any chat with the specified chatid.
+         * - MegaError::API_EACCESS - If the logged in user doesn't have operator privileges
+         *
+         * @param chatid MegaHandle that identifies the chat room
+         * @param period retention timeframe after which all messages in the chat are automatically deleted
```
Comment (sergiohs84) [inline include/megaapi.h:15654]:
```suggestion
         * @param period retention timeframe after which older messages in the chat are automatically deleted
```

File: src/commands.cpp [hunk 4]
```diff
@@ -6539,6 +6539,47 @@ void CommandArchiveChat::procresult()
...
+}
+
+void CommandSetChatRetentionTime::procresult()
+{
+    if (client->json.isnumeric())
+    {
+        error e = (error) client->json.getint();
+        if (e == API_OK)
+        {
+            if (client->chats.find(mChatid) == client->chats.end())
```
Comment (sergiohs84) [inline src/commands.cpp:6566]: Chats cannot be deleted, so I feel this block is not needed. Thoughts?

**Comment by jgandres** [comment]
trigger compilation

**Pull request merged**